x-pack/plugin/wildcard/src/test/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldAggregationTests.java

```diff
@@ -0,0 +1,142 @@
+    }
+
+    private void addFields(ParseContext.Document parseDoc, Document doc, String docContent) throws IOException {
+        ArrayList<IndexableField> fields = new ArrayList<>();
+        wildcardFieldMapper.createFields(docContent, parseDoc, fields);
+
+        for (IndexableField indexableField : fields) {
+            doc.add(indexableField);
+        }
+    }
+
+    private void indexDoc(ParseContext.Document parseDoc, Document doc, RandomIndexWriter iw) throws IOException {
+        IndexableField field = parseDoc.getByKey(wildcardFieldMapper.name());
+        if (field != null) {
+            doc.add(field);
+        }
+        iw.addDocument(doc);
+    }
+
+    private void indexStrings(RandomIndexWriter iw, String... values) throws IOException {
+        Document doc = new Document();
+        ParseContext.Document parseDoc = new ParseContext.Document();
+        for (String value : values) {
+            addFields(parseDoc, doc, value);
+        }
+        indexDoc(parseDoc, doc, iw);
+    }
+
+    public void testTermsAggregation() throws IOException {
+        TermsAggregationBuilder aggregationBuilder = new TermsAggregationBuilder("_name")
+            .field(WILDCARD_FIELD_NAME)
+            .order(BucketOrder.key(true));
+
+        testCase(aggregationBuilder,
+            iwc, new MatchAllDocsQuery(),
+            iw -> {
+                indexStrings(iw, "a");
+                indexStrings(iw, "a");
+                indexStrings(iw, "b");
+                indexStrings(iw, "b");
+                indexStrings(iw, "b");
+                indexStrings(iw, "c");
+            },
+            (Consumer<StringTerms>) result -> {
+                assertTrue(AggregationInspectionHelper.hasValue(result));
+
+                assertEquals(3, result.getBuckets().size());
+                assertEquals("a", result.getBuckets().get(0).getKeyAsString());
+                assertEquals(2L, result.getBuckets().get(0).getDocCount());
+                assertEquals("b", result.getBuckets().get(1).getKeyAsString());
+                assertEquals(3L, result.getBuckets().get(1).getDocCount());
+                assertEquals("c", result.getBuckets().get(2).getKeyAsString());
+                assertEquals(1L, result.getBuckets().get(2).getDocCount());
+            },
+            wildcardFieldType);
+    }
+
+    public void testCompositeTermsAggregation() throws IOException {
+        CompositeAggregationBuilder aggregationBuilder = new CompositeAggregationBuilder(
+            "name",
+            List.of(
+                new TermsValuesSourceBuilder("terms_key").field(WILDCARD_FIELD_NAME)
+            )
+        );
+
+        testCase(aggregationBuilder,
+            iwc, new MatchAllDocsQuery(),
+            iw -> {
+                indexStrings(iw, "a");
+                indexStrings(iw, "c");
+                indexStrings(iw, "a");
+                indexStrings(iw, "d");
+                indexStrings(iw, "c");
```

Comment:
Could you try running with `aggregateAfter` set? I think that'd be useful to make sure works as well.